### PR TITLE
Handle qualifying elements

### LINKED ISSUE #52: Handle Qualifying Elements
```css
h1.example {
  display: block;
  text-align: center;
}
```
should become
```css
h1.rp__display__--COLONblock {
  display: block;
}
h1.rp__text-align__--COLONcenter {
  text-align: center;
}
```
or if uglified
```css
h1.rp__0 {
  display: block;
}
h1.rp__1 {
  text-align: center;
}
```

### PR DESCRIPTION
**Pull request by TheJaredWilcurt**
Coverage Badge

Fixes #52

* TDD (failing tests) set up

### CODE REVIEW

**Comment by gwenf** [comment]
I'm wondering why the test for the script tag on line 876 in `index.test.js` is `.qualifying`. Shouldn't it be `h1.qualifying`?

File: tests/index.test.js [hunk 1]
```diff
@@ -808,6 +813,164 @@ describe('Red Perfume', () => {
...
+                            </body></html>
+                          `, 28));
+
+                        expect(err)
+                          .toEqual(undefined);
+                      }
+                    }
+                  ],
+                  scripts: {
+                    result: function (result, err) {
```
Comment (TheJaredWilcurt) [inline tests/index.test.js:876]: > "I'm wondering why the test for the script tag on line 876 in `index.test.js` is `.qualifying`. Shouldn't it be `h1.qualifying`?"

@gwenf Don't add PR comments, add inline comments, that can be replied to directly, like this.


The JSON output is meant to be used by JavaScript, so you could do something like this:
```html
<div :class="$styles.qualifying"></div>
<h1 :class="$styles.qualifying"></h1>
```
and it would show up in the DOM as:
```html
<div class="rp__border__--COLON1px_____-solid_____-__--OCTOTHORP000 rp__padding__--COLON10px rp__line-height__--COLON1__--DOT4"></div>
<h1 class="rp__border__--COLON1px_____-solid_____-__--OCTOTHORP000 rp__padding__--COLON10px rp__line-height__--COLON1__--DOT4"></h1>
```

But the CSS that actually applies the styles would be:

```css
h1.rp__border__--COLON1px_____-solid_____-__--OCTOTHORP000 { border: 1px solid #000 }
h1.rp__padding__--COLON10px { padding: 10px }
h1.rp__line-height__--COLON1__--DOT4 { line-height: 1.4 }
```

Or something to that effect, the JS portion will change later. Depending on edge cases as we go further, we may need to change these qualifying classes to include the tag name. Like:

```css
h1.h1-rp__padding__--COLON10px { padding: 10px }
```

Comment (gwenf) [reply tests/index.test.js:876]: Oh, I see now. Thanks.

**Comment by gwenf** [comment]
The fs.writeFileSync calls are throwing an error when I'm trying to run the tests because the file systems are for Windows computers: 'C:\out.css'
When I change it to something like `/Users/gwen/code/projects/node/red-perfume/out.css`, it doesn't throw an error.

File: src/process-tasks.js [hunk 2]
```diff
@@ -0,0 +1,110 @@
+const fs = require('fs');
+const css = require('./src/css.js');
+const html = require('./src/html.js');
+const helpers = require('./src/helpers.js');
```
Comment (TheJaredWilcurt) [inline src/process-tasks.js:4]:
```suggestion
const css = require('./css.js');
const html = require('./html.js');
const helpers = require('./helpers.js');
```

File: tests/index.test.js [hunk 3]
```diff
@@ -1,3 +1,5 @@
+/* eslint-disable max-lines-per-function */
+
 const mockfs = require('mock-fs');
```
Comment (TheJaredWilcurt) [inline tests/index.test.js:3]: @gwenf Make sure that all comments are inline so they can be replied to directly. Even if it's just in a random file on a random line.

This file is using `mock-fs`, you can us it as an example. I am not able to reproduce your issue. The current commited code runs identically with `npm t` in Windows and Ubuntu.

**Comment by TheJaredWilcurt** [comment]
Ported to `red-perfume-css` and `red-perfume-html`. Closing this PR.
